### DISCUSSION TITLE
AWS_PROFILE and ~/.aws/credentials

### DISCUSSION THREAD

**Discussion by joebowbeer** [crossplane/crossplane#3246] [General]
Are there plans for Crossplane's aws ProviderConfig to support [sources other than access_key and secret_key environment variables](https://www.padok.fr/en/blog/authentication-aws-profiles)?

The Terraform provider, for comparison, can obtain creds given only an AWS_PROFILE, provided that it can locate the shared credentials file (and shared configuration file for non-default profiles).

https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication-and-configuration

This way, creds managers like [Leapp](https://www.leapp.cloud/) can work seamlessly with Terraform:

https://blog.pethron.me/how-to-access-aws-and-azure-in-terraform

If Crossplane could consume only AWS_PROFILE then it would be possible to use a creds manager like Leapp to refresh the short term credentials.

Should I create a feature request?

**Comment by bobh66** [comment]
The AWS provider can use IRSA to provide access to a specific Role and then use assumeRole in ProviderConfig to switch credentials.

See https://github.com/crossplane-contrib/provider-aws/blob/master/AUTHENTICATION.md#using-iam-roles-for-serviceaccounts

**Comment by joebowbeer** [comment]
@bobh66 What I am looking for is support for AWS_PROFILE in local clusters without IRSA. (I assume IRSA is not an option for local clusters; please correct me if I'm wrong.)

While I have lots of use cases for IRSA in EKS, my use cases motivating this request for AWS_PROFILE are local testing, development and bootstrapping.

I would like to use a credential manager (e.g. Leapp, aws-vault?) to obtain short-term creds and refresh them as needed.

When I see the same `secret_key` / `access_key` instructions and custom scripts pasted into a half dozen articles and tutorials, I can't help thinking there's a better way that might benefit others.

In `kind`, you can mount `~/.aws` to each worker node (container) and the AWS_SDK (called by the pod) can then use that to assume the role based on AWS_PROFILE.

Returning to IRSA, we wonder if `kind` can emulate EKS worker nodes (using an EKS image). For example, on EKS, pods inherit the worker node IAM role by default. This works by the SDK falling back to EC2 metadata when it can't find creds in the env (right?). In theory, then, the kind cluster could emulate the metadata endpoint and provide credentials this way; then Crossplane, using the AWS sdk would "just work"?

**Reply by bobh66** [reply]
@haarchri is much more familiar with https://github.com/crossplane-contrib/provider-aws - might want to move this to an issue in that repo.

**Comment by joebowbeer** [comment]
@haarchri Created RFE https://github.com/crossplane-contrib/provider-aws/issues/1439
